webapp/src/Command/CheckDatabaseConfigurationDefaultValuesCommand.php

```diff
@@ -0,0 +1,100 @@
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Class CheckDatabaseConfigurationDefaultValuesCommand
+ * @package App\Command
+ */
+class CheckDatabaseConfigurationDefaultValuesCommand extends Command
+{
+    /**
+     * @var ConfigurationService
+     */
+    protected $config;
+
+    /**
+     * CheckDatabaseConfigurationDefaultValuesCommand constructor.
+     * @param ConfigurationService $config
+     * @param string|null $name
+     */
+    public function __construct(
+        ConfigurationService $config,
+        string $name = null
+    ) {
+        parent::__construct($name);
+        $this->config = $config;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('domjudge:db-config:check')
+            ->setDescription(
+                'Check if the default values of the database configuration are valid'
+            );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws \Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $style = new SymfonyStyle($input, $output);
+        $messages = [];
+        foreach ($this->config->getConfigSpecification() as $specification) {
+            $message = sprintf(
+                'Configuration %s (in category %s) is of type %s but has wrong type for default_value (%s)',
+                $specification['name'], $specification['category'],
+                $specification['type'], json_encode($specification['default_value'])
+            );
+            switch ($specification['type']) {
+                case 'bool':
+                    if (!is_bool($specification['default_value'])) {
+                        $messages[] = $message;
+                    }
+                    break;
+                case 'int':
+                    if (!is_int($specification['default_value'])) {
+                        $messages[] = $message;
+                    }
+                    break;
+                case 'string':
+                    if (!is_string($specification['default_value'])) {
+                        $messages[] = $message;
+                    }
+                    break;
+                case 'array_val':
+                    if (!empty($specification['default_value']) && (
+                            !is_array($specification['default_value']) ||
+                            !is_int(key($specification['default_value'])))) {
+                        $messages[] = $message;
+                    }
+                    break;
+                case 'array_keyval':
+                    if (!empty($specification['default_value']) && (
+                            !is_array($specification['default_value']) ||
+                            !is_string(key($specification['default_value'])))) {
```

Comment:
The checks for `array_val` and `array_keyval` look wrong to me. How can `$specification['default_value']` be both empty and an array? Shouldn't that be *or*? And the next `||` be `&&`?

Reply:
Maybe my logic is rusty, but what I want to check: if the value is non-empty but it does contain invalid key types, we print a message. That is what I'm doing right?

Reply:
Ah, sorry, my logic is wrong!
Although I think that the logic as written is a bit confusing: it's checking whether the entry is *not* valid. If the check would be written as:
```php
                    if (!(empty($specification['default_value']) || (
                            is_array($specification['default_value']) &&
                            is_int(key($specification['default_value']))))) { ... }
```
then I'd say it's slightly clearer, since now the whole conditional inside the outer `!( ... )` checks whether the entry *is* valid.

Reply:
Changed it to your suggestion :)